src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx

```diff
@@ -60,28 +72,31 @@ export const MultipleAnimationModel: FC<MultipleAnimationModelProps> = ({
 
   useEffect(() => {
     const mixer = mixerRef.current;
+    const prevAction = activeActionRef.current;
     const newClip = loadedAnimations[activeAnimation];
 
     if (!newClip || !mixer) return;
+    if (prevAction && prevAction.getClip().name === newClip.name) return;
 
     const newAction = mixer.clipAction(newClip);
-    const prevAction = activeAction;
 
     if (prevAction) {
       newAction.reset();
       newAction.crossFadeFrom(prevAction, 0.5, true);
     }
 
     newAction.play();
-    setActiveAction(newAction);
-  }, [activeAnimation, loadedAnimations, activeAction, nodes.Armature]);
+    activeActionRef.current = newAction;
+  }, [activeAnimation, loadedAnimations]);
 
   useFrame((state, delta) => {
     mixerRef.current?.update(delta);
+    animationTimeRef.current = activeActionRef.current?.time || 0;
   });
 
   useEmotion(nodes, emotion);
   useIdleExpression('blink', nodes);
+  useFallback(nodes, setModelFallback);
```

Comment:
Introduced the fallback as with all models - debatable if it's actually needed?

Reply:
I don't think it's really needed for all models, didn't see any real usecase of it, especially for editor

Reply:
Updated to use the component, not container